src/sys/socket/addr.rs

```diff
@@ -702,7 +702,7 @@ pub mod sys_control {
             let addr = sockaddr_ctl {
                 sc_len: mem::size_of::<sockaddr_ctl>() as c_uchar,
                 sc_family: AddressFamily::System as c_uchar,
-                ss_sysaddr: consts::AF_SYS_CONTROL as uint16_t,
```

Comment:
I don't think `AF_SYS_CONTROL` should be considered as the same type as other `SA_*` constants. You'll not use it in `socket` function for example, there you should use `AF_SYSTEM`. `AF_SYS_CONTROL` seems like a subtype to `AF_SYSTEM` that only appear in `sockaddr_ctl` when connecting to the socket.
But you are right, `AF_*` constants are used sometimes as `int` (in `socket` function) or as `sa_family_t` (in `sockaddr` struct).
I guess the problem is not applicable to `AF_SYS_CONTROL` but I cannot find good documentation about its use though :/